be2-scala/src/main/scala/ch/epfl/pop/decentralized/Monitor.scala

```diff
@@ -82,9 +85,19 @@ final case class Monitor(
     case ConnectionMediator.Ping() =>
       log.info("Received ConnectionMediator ping")
       connectionMediatorRef = sender()
-      new Thread(new FileMonitor(connectionMediatorRef)).start()
+      fileMonitor = new FileMonitor(this.self)
+      new Thread(fileMonitor).start()
 
-    case _ => /* DO NOTHING */
+    case msg: ConnectionMediator.ConnectTo =>
+      connectionMediatorRef ! msg
```

Comment:
This might deserve a comment to explain the logic behind it has it isn't trivial to understand why it is used by just looking at the code.